modules/importer/src/main/scala/ConfigReader.scala

```diff
@@ -38,72 +51,186 @@ object ConfigReader3 {
     def enum[A: Enumerated: TypeTag](f: A => String): Read[A] =
       cast[String].map(ufindp(f))
 
-    def seq[S <: SequenceableSpType, A: Enumerated: TypeTag](f: A => String): Read[A] =
-      cast[S].map(s => ufindp(f)(s.sequenceValue))
-
     val unit:          Read[Unit         ] = _ => ()
     val string:        Read[String       ] = cast[String]
     val double:        Read[Double       ] = cast[Double]
     val int:           Read[Int          ] = cast[Int]
     val long:          Read[Long         ] = cast[Long]
     val offsetAngle:   Read[Angle        ] = string.map(s => Angle.fromArcsecs(s.toDouble))
-    val offsetP:       Read[OffsetP      ] = offsetAngle.map(OffsetP(_))
-    val offsetQ:       Read[OffsetQ      ] = offsetAngle.map(OffsetQ(_))
-    val instrument:    Read[Instrument   ] = enum(_.tccValue)
     val yesNo:         Read[Boolean      ] = cast[YesNoType].map(_.toBoolean)
     val durSecs:       Read[Duration     ] = double.map(_.toInt).map(Duration.ofSeconds(_))
-    val gcalLamp:      Read[GCalLamp     ] = cast[JSet[Object]].map(_.iterator.next.toString).map(ufindp[GCalLamp, String](_.tccValue))
-    val gcalShutter:   Read[GCalShutter  ] = seq[OldGCal.Shutter, GCalShutter](_.tccValue)
-    val f2fpu:         Read[F2FpUnit     ] = seq[OldF2.FPUnit,      F2FpUnit     ](_.tccValue)
-    val f2filter:      Read[F2Filter     ] = seq[OldF2.Filter,      F2Filter     ](_.tccValue)
-    val f2lyotwheel:   Read[F2LyotWheel  ] = seq[OldF2.LyotWheel,   F2LyotWheel  ](_.tccValue)
-    val f2disperser:   Read[F2Disperser  ] = seq[OldF2.Disperser,   F2Disperser  ](_.tccValue)
-
-    // It appears that the window cover is sometimes "bare" and someimes wrapped in `Some` ... it is
-    // not `None` in my test data so there's not a mapping for it.
-    val f2windowcover: Read[F2WindowCover] = { a =>
-      ufindp[F2WindowCover, String](_.tccValue) {
-        a match {
-          case a: OldF2.WindowCover        => a.sequenceValue
-          case s: GSome[OldF2.WindowCover] => s.getValue.sequenceValue
-        }
+  }
+
+
+  sealed abstract class System(private val system: String) {
+
+    override def toString: String =
+      s"System($system)"
+
+    final class Key[A](val name: String, val write: Show[A], val read: Read[A])(implicit ev: TypeTag[A]) {
+      val path = s"$system:$name"
+
+      def legacyGet(c: Map[String, AnyRef]): Option[AnyRef] =
+        c.get(path)
+
+      override def toString: String =
+        s"Key[${clean(ev.tpe.toString)}]($path)"
+
+      final class Entry private[Key](a: A) {
+        val value = a
+        val key   = Key.this
+        override def toString: String =
+          s"Key.Entry[${clean(ev.tpe.toString)}]($path -> ${write(a)})"
+      }
+
+      def apply(a: A) = new Entry(a)
+    }
+
+    protected object Key {
+      def apply[A: TypeTag](name: String, write: Show[A])(read: Read[A]): Key[A] =
```

Comment:
I anticipate that the `Read` implementations will be involved in some cases and so it will be more convenient to write them this way.